dolphinscheduler-dao/src/main/resources/sql/dolphinscheduler_mysql.sql

```diff
@@ -790,6 +790,22 @@ CREATE TABLE `t_ds_resources` (
 -- Records of t_ds_resources
 -- ----------------------------
 
+-- ----------------------------
+-- Table structure for t_ds_relation_resources_task
+-- ----------------------------
+DROP TABLE IF EXISTS `t_ds_relation_resources_task`;
+CREATE TABLE `t_ds_relation_resources_task` (
+  `id` int NOT NULL AUTO_INCREMENT COMMENT 'key',
+  `full_name` varchar(255) DEFAULT NULL,
+  `type` tinyint DEFAULT NULL COMMENT 'resource type,0:FILE,1:UDF',
+  PRIMARY KEY (`id`),
+  UNIQUE KEY `t_ds_relation_resources_task_un` (`full_name`,`type`)
+) ENGINE=InnoDB AUTO_INCREMENT=1 DEFAULT CHARSET=utf8mb3
```

Comment:
I think this table is about resources, not the relation between task and resources.
Why not create this table like this, and then we can remove the `resource_ids_new` field in `t_ds_task_definition`
```suggestion
CREATE TABLE `t_ds_relation_resources_task` (
  `id` int NOT NULL AUTO_INCREMENT COMMENT 'key',
  `task_id` int(11) DEFAULT NULL COMMENT 'task id',
  `full_name` varchar(255) DEFAULT NULL,
  `type` tinyint DEFAULT NULL COMMENT 'resource type,0:FILE,1:UDF',
  PRIMARY KEY (`id`),
  UNIQUE KEY `t_ds_relation_resources_task_un` (`task_id`, `full_name`,`type`)
) ENGINE=InnoDB AUTO_INCREMENT=1 DEFAULT CHARSET=utf8mb3
```

Reply:
> I think this table is about resources, not the relation between task and resources. Why not create this table like this, and then we can remove the `resource_ids_new` field in `t_ds_task_definition`

It is a relation between tasks and resources because we store resources into this table only when a task uses them.

Having a `task_id` field in table definitely helps a lot! I will make some changes based on this new field.

Reply:
@caishunfeng In the latest commit, I added the `task_id` field. While testing it, I found there was no logic in updating task definition-related tables in the repo. For example, In the current implementation, the modification to resources (rename, delete) used by a task won't update the task definition log, which it should. Therefore, at the end I used `processDefinitionService.updateProcessDefinition()` to implement the logic.